lib/codeowners/cli/main.rb

```diff
@@ -35,16 +37,36 @@ def check(repo = '.')
       desc 'config', 'Checks config is consistent or configure it'
       subcommand 'config', Codeowners::Cli::Config
 
+      # interactive set to true as this may ask for organization/github token
+      option :interactive, default: true, type: :boolean, hide: true
+      desc 'generate REPO', 'Generates .github/team.txt based on github organization'
+      def generate(repo = '.')
+        @repo = repo
+        owners = owners_from_github
+        setup_checker
+        @checker.owners_list = owners
+        @checker.persist_owners_list!
+      end
+
       private
 
       def interactive_mode
         @checker.fix!
-        return unless @codeowners_changed
+        return unless @content_changed
 
         write_codeowners
         @checker.commit_changes! if options[:autocommit] || yes?('Commit changes?')
       end
 
+      def owners_from_github
+        organization = ENV['GITHUB_ORGANIZATION']
+        organization ||= ask('GitHub organization (e.g. github): ')
+        token = ENV['GITHUB_TOKEN']
+        token ||= ask('Enter GitHub token: ', echo: false)
```

Comment:
I've suggested separate file to handle owners, we should move the above logic there.